inputremapper/injection/mapping_handlers/combination_handler.py

```diff
@@ -109,22 +111,26 @@ def notify(
         # maybe this changes the activation status (triggered/not-triggered)
         is_activated = self.is_activated()
 
-        if is_activated == was_activated or is_activated == self._output_state:
+        if is_activated == self._output_active:
             # nothing changed
-            if self._output_state:
-                # combination is active, consume the event
-                return True
+            if is_pressed:
+                # If the output is active, a key-down event triggered it, which then
+                # did not get forwarded, therefore it doesn't require a release.
+                self.require_release_later(not self._output_active, event)
+                # output is active: consume the event
+                # output inactive: forward the event
+                return self._output_active
             else:
-                # combination inactive, forward the event
-                return False
+                # Else if it is released: Returning `False` means that the event-reader
+                # will forward the release.
+                return not self.should_release_event(event)
 
         if is_activated:
-            # send key up events to the forwarded uinput
-            self.forward_release()
             event = event.modify(value=1)
         else:
-            if self._output_state or self.mapping.is_axis_mapping():
-                # we ignore the suppress argument for release events
+            # If not activated. All required keys are not yet pressed.
+            if self._output_active or self.mapping.is_axis_mapping():
+                # we ignore the `suppress` argument for release events
```

Comment:
this is always true because
`is_activated != self._output_active` and `is_activated == False`

Reply:
good catch